src/Lean/Meta/Tactic/Simp/Rewrite.lean

```diff
@@ -387,9 +387,10 @@ def simpGround : Simproc := fun e => do
   if ctx.simpTheorems.isErased (.decl declName) then return .continue
   -- Matcher applications should have been reduced before we get here.
   if (← isMatcher declName) then return .continue
-  trace[Meta.Tactic.Simp.ground] "seval: {e}"
-  let r ← seval e
-  trace[Meta.Tactic.Simp.ground] "seval result: {e} => {r.expr}"
+  let r ← withTraceNode `Meta.Tactic.simp.ground (fun
```

Comment:
@leodemoura Are you happy with this nesting adjustment? This PR only affects the display of nested traces as on the top level they can be mixed with other messages, which would effectively require a completely separate implementation.

Reply:
In action after fixing the `withTraceNode` bug: 
![image](https://github.com/leanprover/lean4/assets/109126/8394ce14-b861-42f9-95c4-9ba249645b50)

Reply:
Yes.